documents/int256.go

```diff
@@ -62,46 +49,24 @@ func NewInt256(n string) (*Int256, error) {
 	return &Int256{*nn}, nil
 }
 
-// Int256FromBytes converts the a big endian byte slice to an Int256
+// Int256FromBytes converts the a big endian 2's compliment byte slice to an Int256
 func Int256FromBytes(b []byte) (*Int256, error) {
 	if len(b) != 32 {
 		return nil, errors.NewTypedError(ErrInvalidInt256, errors.New("value: %x", b))
 	}
 
-	// check and record the sign bit
-	neg := false
-	if byteutils.IsBitSet(b[signByteIdx], 7) {
-		neg = true
-	}
-	// ignore the sign bit
-	b[signByteIdx] = byteutils.ClearBit(b[signByteIdx], 7)
-
-	nn := new(big.Int)
-	nn.SetBytes(b[:])
-	if !isValidInt256(*nn) {
-		return nil, errors.NewTypedError(ErrInvalidInt256, errors.New("value: %s", nn.String()))
-	}
-
-	// negative
-	if neg {
-		nn = nn.Neg(nn)
-	}
-
+	nn := math.S256(new(big.Int).SetBytes(b))
 	return &Int256{*nn}, nil
 }
 
-// Bytes returns the big endian 32 byte representation of this int256. First bit of LSB(least significant byte) is used as the sign bit.
-func (i *Int256) Bytes() [32]byte {
+// Bytes returns the big endian 2's compliment 32 byte representation of this int256.
+func (i Int256) Bytes() [32]byte {
+	v := new(big.Int).Mul(&i.v, big.NewInt(1))
```

Comment:
Why do we multiply by 1, is it only for rounding?

Reply:
U256 is a destructive operation. We need to clone the value so that it is usable after calling bytes. Hence we clone it using a multiplier and shouldn't affect any rounding since we don't support any decimal here